### PR TITLE
Added ES6 feature support to NodeJS-Request codegen

### PR DESCRIPTION
**Pull request by sastava007**
**What it does**
It allows the user to generate the code snippet for NodeJS Request in ES6 format.
This is in response to issue #115 requested by one of the users.

**Changes I made**

- Registered a new parameter ES6_enabled in **`options()`** which allows the user to toggle between whether codegen generates snippet with ES6 features or not.

- By default ES6_enabled is set to `false`

- Also added the new created option parameter in **`structure.test.js`** file to register it as a valid option id.

- Added unit tests to verify the generated snippets.

**Follow Up**
This PR is first in the series of upcoming PRs which will add the feature of generating code snippet in ES6 notation with NodeJS.

### CODE REVIEW

**Comment by sastava007** [comment]
@shreys7 I have made all the changes as you said in your last review, hope it's fine now.

File: codegens/nodejs-request/lib/request.js [hunk 1]
```diff
@@ -13,9 +13,15 @@ var _ = require('./lodash'),
  * @returns {String} - nodejs(request) code snippet for given request object
  */
 function makeSnippet (request, indentString, options) {
-  var snippet = 'var request = require(\'request\');\n',
+  var snippet,
     optionsArray = [],
     isFormDataFile = false;
+  if (options.ES6_enabled) {
```
Comment (shreys7) [inline codegens/nodejs-request/lib/request.js]: This is your code
```
if (options.ES6_enabled) {
    snippet = 'let request = require(\'request\');\n';
  }
  else {
    snippet = 'var request = require(\'request\');\n';
  }
```

It should be more like the one mentioned below(lesser code repetition):
```
if (options.ES6_enabled) {
    snippet = 'let ';
  }
  else {
    snippet = 'const ';
  }
snippet += 'request = require(\'request\');\n'
```

File: codegens/nodejs-request/lib/request.js [hunk 2]
```diff
@@ -13,9 +13,15 @@ var _ = require('./lodash'),
...
+    snippet = 'let request = require(\'request\');\n';
```
Comment (shreys7) [inline codegens/nodejs-request/lib/request.js]: `const` instead of `let`

File: codegens/nodejs-request/lib/request.js [hunk 3]
```diff
@@ -24,9 +30,19 @@ function makeSnippet (request, indentString, options) {
     });
   }
   if (isFormDataFile) {
-    snippet += 'var fs = require(\'fs\');\n';
+    if (options.ES6_enabled) {
```
Comment (shreys7) [inline codegens/nodejs-request/lib/request.js]: same comment as above. Combine for lesser code repetition

File: codegens/nodejs-request/lib/request.js [hunk 4]
```diff
@@ -24,9 +30,19 @@ function makeSnippet (request, indentString, options) {
...
+      snippet += 'let fs = require(\'fs\');\n';
```
Comment (shreys7) [inline codegens/nodejs-request/lib/request.js]: `const` instead of `let`

Comment (sastava007) [reply codegens/nodejs-request/lib/request.js]: @shreys7 Last time, I used const so you asked me to use var/ let. That's why I changed.

Comment (shreys7) [reply codegens/nodejs-request/lib/request.js]: no. Those changes for if-else statements. We should use const for fs and request.

**Comment by sastava007** [comment]
@shreys7 Have updated as you said, once this one gets accepted for **`request`** I'll fix the **`native`** and **`unirest`** with the same approach.

**Comment by sastava007** [comment]
@umeshp7 Can you please review this?

File: codegens/nodejs-request/lib/request.js [hunk 5]
```diff
@@ -120,6 +145,13 @@ function getOptions () {
       type: 'boolean',
       default: false,
       description: 'Remove white space and additional lines that may affect the server\'s response'
+    },
+    {
+      name: 'Generate the code snippet with ES6 features',
+      id: 'ES6_enabled',
+      type: 'boolean',
+      default: false,
+      description: 'Allows the user to generate code snippet with latest EcmaScript 6 (ES6) features'
```
Comment (umeshp7) [inline codegens/nodejs-request/lib/request.js]: This description is user-facing. So we don't need to specify 'user'. We have to come up with something that the user will read. 
We could go with something like: "Modifies code snippet to incorporate ES6 (EcmaScript) features."

File: codegens/nodejs-request/lib/request.js [hunk 6]
```diff
@@ -120,6 +145,13 @@ function getOptions () {
       type: 'boolean',
       default: false,
       description: 'Remove white space and additional lines that may affect the server\'s response'
+    },
+    {
+      name: 'Generate the code snippet with ES6 features',
```
Comment (umeshp7) [inline codegens/nodejs-request/lib/request.js]: This could be simply 'Enable ES6 features'.

**Comment by umeshp7** [comment]
We'll need an extra set of newman tests, with this option enabled true. We need to make sure the snippet generated is runnable.

**Comment by sastava007** [comment]
@umeshp7 I have added a new block inside the main mocha suite with **`ES6_enabled`** : `true`, please have a look.

**Comment by shreys7** [comment]
@sastava007 Merging this, you can start working on `native` and `unirest`. Thanks for the contribution.

**Pull request merged**